saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/RelyingPartyRegistration.java

```diff
@@ -299,12 +300,14 @@ public static Builder withRelyingPartyRegistration(RelyingPartyRegistration regi
 		private final String entityId;
 		private final boolean wantAuthnRequestsSigned;
 		private final String singleSignOnServiceLocation;
+		private final String nameIdFormat;
```

Comment:
NameID is a part of the IDPSSODescriptor

Reply:
I believe it's in both. You can see it listed for SPSSODescriptor in [this metadata overview document](https://www.oasis-open.org/committees/download.php/51890/SAML%20MD%20simplified%20overview.pdf) on page 8, section 2.14.

There may be value in adding this in both places, but since we're generating an SPSSODescriptor for this ticket, let's please focus on that use case.